### PR TITLE
Implement tenant directory deletion on tenant deletion

### PR DESCRIPTION
**Pull request by sumedhe**
## Purpose
This will delete the tenant directory located under `${CARBON_HOME}/repository/tenants/{TenantID}` in the tenant deletion process.

Fixes: #194

### CODE REVIEW

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java [hunk 1]
```diff
@@ -605,6 +609,27 @@ public static void deleteProductSpecificTenantData(String dataSourceName, String
...
+            log.debug("Deleting tenant directory of tenant: " + tenantId);
+        }
+
+        String tenantDirPath = CarbonUtils.getCarbonTenantsDirPath() + File.separator + tenantId;
+        File tenantDir = new File(tenantDirPath);
+
+        try {
+            FileUtils.deleteDirectory(tenantDir);
+        } catch (IOException e) {
+            log.error("Error in deleting tenant directory: " + tenantDirPath);
```
Comment (omindu) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java:629]: add `e` to the log

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java [hunk 2]
```diff
@@ -20,6 +20,7 @@
 import org.apache.axis2.context.ConfigurationContext;
 import org.apache.axis2.engine.AxisConfiguration;
 import org.apache.commons.dbcp.BasicDataSource;
+import org.apache.commons.io.FileUtils;
```
Comment (omindu) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java:23]: Make sure this is imported as an OSGi import if already haven't

**Comment by abhi4u1947** [comment]
There are other places too where {TenantID} related directory get created, those also needs to be deleted.

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java [hunk 3]
```diff
@@ -196,14 +196,14 @@ public String addSkeletonTenant(TenantInfoBean tenantInfoBean) throws Exception
...
+            tenantId = TenantMgtServiceComponent.getTenantManager().getTenantId(tenantInfoBean.getTenantDomain());
         } catch (org.wso2.carbon.user.api.UserStoreException e) {
             String msg = "Error in getting tenant id";
             log.error(msg, e);
             throw new Exception(msg, e);
         }
-        if(tenantId<0){
-            String msg = "Tenant is not added in user store "+tenantInfoBean.getTenantDomain();
+        if (tenantId < 0) {
+            String msg = "Tenant is not added in user store " + tenantInfoBean.getTenantDomain();
```
Comment (ashensw) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java:206]: Is this msg correct?

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java [hunk 4]
```diff
@@ -291,13 +291,14 @@ public String addSkeletonTenant(TenantInfoBean tenantInfoBean) throws Exception
     }
 
     /**
-     * Method to retrieve all the partial search domain tenants paginated
+     * Method to retrieve all the partial search domain tenants paginated.
      *
      * @param pageNumber Number of the page.
      * @return PaginatedTenantInfoBean
      * @throws Exception if failed to getTenantManager;
```
Comment (ashensw) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java:298]: Is this semicolon needed here?

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java (hunk 3 above)
Comment (sumedhe) [reply components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java:206]: I hope so. It seems the message is shown when the tenant is the super-tenant. Any suggestion?

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java (hunk 4 above)
Comment (sumedhe) [reply components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java:298]: Fixed in #207

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java (hunk 3 above)
Comment (sumedhe) [reply components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java:206]: Fixed in #207

**Pull request merged**